### PR TITLE
Trip sharing

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Provides a new endpoint to query a given user's monitored trips, including trips on which a user is a companion and or an observer. The field type for `primary` user is changed to `MobilityProfileLite` as a result from introducing the endpoint in #266 for querying the dependent profile.

Out of scope: email notifications to users added as companions/observers/primary travelers will be addressed in a separate PR.

### CODE REVIEW

**Review by josh-willis-arcadis — APPROVED** [review]
LGTM!

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java [hunk 1]
```diff
@@ -176,7 +176,7 @@ public class MonitoredTrip extends Model {
      */
     public boolean notifyAtLeadingInterval = true;
 
-    public RelatedUser primary;
+    public MobilityProfileLite primary;
     public RelatedUser companion;
     public List<RelatedUser> observers = new ArrayList<>();
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java:181]: Probably out of scope, but should any updates to related users be reflected in monitored trips? For instance, if a related user is removed?

File: src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java [hunk 2]
```diff
@@ -53,6 +54,16 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
         super.buildEndpoint(modifiedEndpoint);
     }
 
+    @Override
+    protected Bson getEntityFilter(OtpUser user) {
+        return Filters.or(
+            Filters.eq(USER_ID_PARAM, user.id),
+            Filters.eq("primary.userId", user.id),
+            Filters.eq("companion.email", user.email),
+            Filters.eq("observers.email", user.email)
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java:63]: I wasn't sure if this syntax would work given that it is an array. Did a quick test and it does!

**Review by br648 — COMMENTED** [review]
Just one comment related to the relationship between related users and monitored trips. Also, I was expecting a new end point, but I guess that wasn't needed in the end?

**Comment by binh-dam-ibigroup** [comment]
> Also, I was expecting a new end point, but I guess that wasn't needed in the end?

Indeed!

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java (hunk 1 above)
Comment (br648) [reply src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java:181]: @binh-dam-ibigroup just this to potentially address if you think it is in scope/important?

Comment (br648) [reply src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java:181]: > @binh-dam-ibigroup just this to potentially address if you think it is in scope/important?

@binh-dam-ibigroup I just need your thoughts on this and then I can approve if it is out of scope.

Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java:181]: @br648 If you are ok with this, we can do delete propagation in another PR.

Comment (br648) [reply src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java:181]: @binh-dam-ibigroup I think it is out of scope. Will approve.

**Review by br648 — APPROVED** [review]
Approving but delete propagation needs to be covered in another PR.

**Pull request merged**